include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,352 @@
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the container.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator end() const noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+    /*!\brief Returns the number of elements in the view.
+     * \returns The number of elements in the view.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     */
+    size_type size()
+    {
+        return urange.size() + (std::floor(urange.size() / step_size) * inserted_range.size());
+    }
+
+    //!\copydoc size()
+    size_type size() const
+    {
+        return urange.size() + (std::floor(urange.size() / step_size) * inserted_range.size());
+    }
+
+    /*!\brief Return the i-th element.
+     * \param i The element to retrieve.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    reference operator[](size_type const i)
+    {
+        size_t combined_size = step_size + inserted_range.size();
+        assert(i < size());
+        if (i % (combined_size) < step_size)
+            return urange[i - (std::floor(i/(combined_size)) * inserted_range.size())];
+        else
+            return inserted_range[(i % (combined_size)) - step_size];
+    }
+
+    //!\copydoc operator[]()
+    const_reference operator[](size_type const i) const
+    {
+        size_t combined_size = step_size + inserted_range.size();
+        assert(i < size());
+        if (i % (combined_size) < step_size)
+            return urange[i - (std::floor(i/(combined_size)) * inserted_range.size())];
+        else
+            return inserted_range[(i % (combined_size)) - step_size];
+    }
+};
```

Comment:
The class still needs a cast to container type. You can probably copy'n'paste this from one of the other views.

Reply:
Hmmmm, I'm not sure what cast to container type you're referring to?